cpp/test/ome-bioformats/metadatatools.cpp

```diff
@@ -553,6 +558,128 @@ TEST_P(CorrectionTest, ValidateAndCorrectModel)
   }
 }
 
+struct ModelTestParameters
+{
+  path file;
+};
+
+template<class charT, class traits>
+inline std::basic_ostream<charT,traits>&
+operator<< (std::basic_ostream<charT,traits>& os,
+            const ModelTestParameters& p)
+{
+  return os << p.file;
+}
+
+namespace
+{
+
+  std::vector<ModelTestParameters>
+  find_model_tests()
+  {
+    std::vector<ModelTestParameters> params;
+
+    ome::xml::OMETransformResolver tr;
+    std::set<std::string> versions = tr.schema_versions();
+
+    path samplesdir(PROJECT_SOURCE_DIR "/components/specification/samples");
+    if (exists(samplesdir) && is_directory(samplesdir))
+      {
+        for (directory_iterator si(samplesdir); si != directory_iterator(); ++si)
+          {
+            if (versions.find(si->path().filename().string()) == versions.end())
+              continue; // Not a schema directory with transforms.
+            path schemadir(si->path());
+            if (exists(schemadir) && is_directory(schemadir))
+              {
+                for (directory_iterator fi(schemadir); fi != directory_iterator(); ++fi)
+                  {
+                    ModelTestParameters p;
+                    p.file = *fi;
+
+                    // 2003-FC/Iron-Plate.ome.tiff.xml
+                    if (schemadir.filename() == path("2003-FC") &&
+                        p.file.filename() == path("Iron-Plate.ome.tiff.xml"))
+                      continue;
+                    // 2003-FC/single-channel.ome.tif.xml
+                    if (schemadir.filename() == path("2003-FC") &&
+                        p.file.filename() == path("single-channel.ome.tif.xml"))
+                      continue;
+                    // 2008-09/6x4y1z1t1c8b-swatch-instrument.ome
+                    if (schemadir.filename() == path("2008-09") &&
+                        p.file.filename() == path("6x4y1z1t1c8b-swatch-instrument.ome"))
+                      continue;
+                    // 2010-06/one-of-everything.xml - parse error
+                    if (schemadir.filename() == path("2010-06") &&
+                        p.file.filename() == path("one-of-everything.xml"))
+                      continue;
+                    // 2010-06/one-of-everything-no-problems.xml - parse error
+                    if (schemadir.filename() == path("2010-06") &&
+                        p.file.filename() == path("one-of-everything-no-problems.xml"))
+                      continue;
+                    // 2013-06/2013-06-datetests.ome - Contains non-POSIX timestamps.
```

Comment:
They are valid according to the schema, but internally inconsistent for the most part.  For example, there's a discrepancy between sizeC and samplesPerPixel for each channel.  The actual failing function is this:

https://github.com/openmicroscopy/bioformats/blob/develop/cpp/lib/ome/bioformats/MetadataTools.cpp#L358

These are consistency checks above and beyond schema validation (it can also optionally correct them).  This is based on corrections applied in the Java code, though it might not be in a single function as here.  Ideally, the samples would be sufficiently correct to pass this validation as well.